suppliers/src/UI/embc-supplier/src/app/app.component.ts

```diff
@@ -28,10 +29,16 @@ export class AppComponent{
   }
 
   ngOnInit() {
+    // This is set in constructor
+    if (!this.supplierService.getServerConfig())
+      this.supplierService.setServerConfig(this.supplierHttp.getServerConfig());
+      
     this.bannerSubscription = this.supplierService.getServerConfig().subscribe(config => {
       this.noticeMsg = config.noticeMsg;
       this.maintMsg = config.maintMsg;
       this.siteDown = config.siteDown;
+    }, err => {
+      console.log(err);
```

Comment:
> 
> 
> Not sure why the config service was renamed to auth.config service

Joseph and I discussed this. This was a new config service added just for authentication, and it's separate from the location where the actual ConfigController is being accessed. This name is more accurate.

That's a good point about the better error-handling. Those lines were added during the experimentation to try and reduce the number of calls to the Config API, but didn't wind up improving things. This would a failure state on displaying a "page will be down for maintenance" banner, so it would probably only confuse users to display a message for something they wouldn't be expecting to see in the first place.

Since this error isn't tied to a user action and isn't likely to impact functionality, I think it should either remain as a console log, or this error should be removed and handled by some sort of application-level error tracking, if it exists.

Reply:
I would recommend consolidating the config API to a single call and tie it to the app component as it is done in responders portal, and to introduce a general error page to handle this types of exceptions, as this can happen if the API is not available - there is no point in continuing loading the site if the config failed to load IMHO.

Reply:
Sounds good. I'll work on better error handling and refactoring the config work into one file. Can this still be merged? It has to be tested against the actual OpenShift systems in dev and test, and we're hoping it can be included in the deployment on Friday.

Reply:
For sure, I already approved it, you can merge it any time you want

Reply:
I don't have permissions to, only someone with write access